Move the apply_tensor_props into its own function for more public use

Gamrix opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #67786

Differential Revision: [D32175962](https://our.internmc.facebook.com/intern/diff/D32175962)

Gamrix committed:
Move the apply_tensor_props into its own function for more public use

[ghstack-poisoned]

Gamrix committed:
Update on "Move the apply_tensor_props into its own function for more public use"




[ghstack-poisoned]

Gamrix committed:
Update on "Move the apply_tensor_props into its own function for more public use"




[ghstack-poisoned]

Gamrix committed:
Update on "Move the apply_tensor_props into its own function for more public use"




[ghstack-poisoned]

Gamrix commented:
@gamrix has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32175962).

Gamrix committed:
Update on "Move the apply_tensor_props into its own function for more public use"


Differential Revision: [D32175962](https://our.internmc.facebook.com/intern/diff/D32175962)

[ghstack-poisoned]

Gamrix commented:
@gamrix has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32175962).

Gamrix committed:
Update on "Move the apply_tensor_props into its own function for more public use"


Differential Revision: [D32175962](https://our.internmc.facebook.com/intern/diff/D32175962)

[ghstack-poisoned]

Gamrix committed:
Update on "Move the apply_tensor_props into its own function for more public use"


Differential Revision: [D32175962](https://our.internmc.facebook.com/intern/diff/D32175962)

[ghstack-poisoned]

Gamrix committed:
Update on "Move the apply_tensor_props into its own function for more public use"


Differential Revision: [D32175962](https://our.internmc.facebook.com/intern/diff/D32175962)

[ghstack-poisoned]

Krovatkin commented on torch/jit/_static_analysis.py:
```diff
@@ -0,0 +1,43 @@
+from torch._C import Graph
+
+
+def apply_input_props_using_example(graph: Graph, example_input: List[Any]):
```
This helper only sets dtype and sizes. There are other properties such as strides, device, undefined or not, requires_grad or not. I would rename this or split into two smaller helpers `apply_sizes`, `apply_dtypes`.

Gamrix replied:
How do I get a `TensorType` object from a tensor in Python?

Krovatkin reviewed (commented):
I would consult w/ @gmagogsfm and @eellison if there might be a more appropriate file for this helper(s) .

Gamrix committed:
Update on "Move the apply_tensor_props into its own function for more public use"


Differential Revision: [D32175962](https://our.internmc.facebook.com/intern/diff/D32175962)

[ghstack-poisoned]

Gamrix committed:
Update on "Move the apply_tensor_props into its own function for more public use"


Differential Revision: [D32175962](https://our.internmc.facebook.com/intern/diff/D32175962)

[ghstack-poisoned]

Gamrix commented:
I messaged @gmagogsfm and updated the location of this file based on his feedback.

Gamrix committed:
Update on "Move the apply_tensor_props into its own function for more public use"


Differential Revision: [D32175962](https://our.internmc.facebook.com/intern/diff/D32175962)

[ghstack-poisoned]

Krovatkin reviewed (approved):
:shipit:

Gamrix committed:
Update on "Move the apply_tensor_props into its own function for more public use"


Differential Revision: [D32175962](https://our.internmc.facebook.com/intern/diff/D32175962)

[ghstack-poisoned]

Gamrix commented:
@gamrix has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32175962).

Gamrix committed:
Update on "Move the apply_tensor_props into its own function for more public use"


Differential Revision: [D32175962](https://our.internmc.facebook.com/intern/diff/D32175962)

[ghstack-poisoned]

Gamrix committed:
Update on "Move the apply_tensor_props into its own function for more public use"


Differential Revision: [D32175962](https://our.internmc.facebook.com/intern/diff/D32175962)

[ghstack-poisoned]
